src/v3/src/components/Widget/index.tsx

```diff
@@ -67,12 +70,22 @@ export const Widget: FunctionComponent<WidgetProps> = (widgetProps) => {
 
   const TERMINAL_STATUSES = [IdxStatus.TERMINAL, IdxStatus.SUCCESS];
   const [data, setData] = useState({});
-  const [messages, setMessages] = useState<IdxMessage[]>([]);
+  const [message, setMessage] = useState<IdxMessage | undefined>();
```

Comment:
I made a change here messages at the widget/index layer is reserved for custom alert messages. Previously, we had multiple places that displayed messages from the IDX Api, it was hard to maintain and know when which logic would display a message vs another. 

Now, there is one message transformer that handles `all` messages from IDX API, and this top level message state handles adhoc messages that can be displayed from components (i.e. client side validation alert message or WebAuthN browser error).

Reply:
can it be moved into formbag, so it's just one of results from the transformation layer? and looks like this change belongs to a separate PR

Reply:
For an adhoc message/alert to be displayed? This probably shouldnt go into them formBag because the message that would be displayed at this level isnt dependent on the transaction or subsequent transformations that accompanies updating the formBag. 

For Example. 

If there is a browser error from WebAuthN, the browser gives back an error response, and we just display that error in an alert box. I can't see a scenario where we would want to set this into the form bag because its totally unrelated to the transaction that the form is built from.